### DISCUSSION TITLE
Compiling a single example at a time without a docker

### DISCUSSION THREAD

**Discussion by amb1s1** [cilium/ebpf#939] [Q&A]
This weeknd is being a struggle trying to get a test environment  to test ebpf. I finally got a docker to compile  a ebpf app. Now I want to go and start testing the examples in the cillium repo. The explanation of how to compile the examples are either not clear or I'm just esxauste of tryin to get this working. 

Can anybody tell me how to do the following: compile a single example: let say the https://github.com/cilium/ebpf/tree/master/examples/kprobe in my own docker and without the use of another docker. Thanks in advance.

**Comment by ti-mo** [comment]
Please read [the README](https://github.com/cilium/ebpf/blob/master/examples/README.md). Running the examples only requires a Go toolchain, no clang needed.

From a cursory reading of the Makefile in the repo root: `container-all` is the default target that runs `make all` in a container. If you have the right toolchain available in your current shell, `make all` should do what you want.
